src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -307,11 +307,16 @@ public Task<Compilation> GetCompilationAsync(SolutionState solution, Cancellatio
                 }
                 else
                 {
-                    return GetOrBuildCompilationInfoAsync(solution, lockGate: true, cancellationToken: cancellationToken)
-                        .ContinueWith(t => t.Result.Compilation, cancellationToken, TaskContinuationOptions.ExecuteSynchronously | TaskContinuationOptions.OnlyOnRanToCompletion, TaskScheduler.Default);
+                    return GetCompilationSlowAsync(solution, cancellationToken);
```

Comment:
Thank you for pointing out the pattern. I searched the whole code for `ExecuteSynchronously` and `OnlyOnRanToCompletion`. Then I found a same bug and fixed it in https://github.com/dotnet/roslyn/pull/19620/commits/73477c501c300acd253063897580e8d6d7fd023b .